### PR TITLE
Remove webchat hardcoding in js

### PR DESCRIPTION
**Pull request by mcgoooo**
This removes the webchat hardcoding in urls in javascript, and also abstracts out the egain specific code. 

it also makes use of data attributes for the open and close urls in preparation for moving this to govspeak or similar, and making it a govuk component.

### CODE REVIEW

File: app/assets/javascripts/webchat/library.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    function apiError () {
+      clearInterval(intervalID)
+      advisorStateChange('ERROR')
+    }
+
+    function advisorStateChange (state) {
+      var currentState = $el.find("." + webchatStateClass + state.toLowerCase())
+      $el.find('[class^="' + webchatStateClass + '"]').addClass('hidden')
+      currentState.removeClass('hidden')
+      GOVUK.analytics.trackEvent('webchat', state)
```
Comment (mcgoooo) [inline app/assets/javascripts/webchat/library.js]: Might need to be lowercase.

File: app/assets/javascripts/webchat.js [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
 var $ = window.$
 
+
+EGAIN_NORMALISATION = {
+  0: "AVAILABLE",
+  1: "UNAVAILABLE",
+  2: "BUSY"
+}
+
+var normalise = function(res){
```
Comment (mcgoooo) [inline app/assets/javascripts/webchat.js]: Need a global test to check eGain gets normalised as we expected.

File: app/assets/javascripts/webchat/library.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      "UNAVAILABLE",
+      "AVAILABLE",
+      "ERROR"
+    ]
+    var $el                 = $(options.$el)
+    var openUrl             = $el.attr('data-open-url')
+    var availabilityUrl     = $el.attr('data-availability-url')
+    var $openButton         = $el.find('.js-webchat-open-button')
+    var webchatStateClass   = 'js-webchat-advisers-'
+    var responseNormaliser  = options.responseNormalisation
```
Comment (mcgoooo) [inline app/assets/javascripts/webchat/library.js]: Maybe responseNormalisation -> responseNormaliser?

File: app/assets/javascripts/webchat.js (hunk 2 above)
Comment (mcgoooo) [inline app/assets/javascripts/webchat.js]: done

File: app/assets/javascripts/webchat/library.js (hunk 1 above)
Comment (mcgoooo) [inline app/assets/javascripts/webchat/library.js]: done

File: app/assets/javascripts/webchat/library.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
+(function (global) {
+  'use strict'
+  var $ = global.jQuery
+  var windowOpen = global.open
+  if (typeof global.GOVUK === 'undefined') { global.GOVUK = {} }
+  var GOVUK = global.GOVUK
+
+
+  function Webchat (options) {
+    var POLL_INTERVAL = 30 * 1000
```
Comment (nickcolley) [inline app/assets/javascripts/webchat/library.js]: Should we change this to 30s rather than 15s?

**Comment by nickcolley** [comment]
We should make sure the commits are in line with https://github.com/alphagov/styleguides/blob/master/git.md

**Pull request merged**